Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs

```diff
@@ -282,6 +283,78 @@ public ActionResult ReadIpsContaining(string value)
             return this.Json(ips, JsonRequestBehavior.AllowGet);
         }
 
+        public void MoveSubmission(int categoryId)
+        {
+            ContestCategory category = this.Data.ContestCategories.GetById(categoryId);
+
+            foreach (var categoryContest in category.Contests)
+            {
+                var competeOnlyParticipants = categoryContest
+                .Participants
+                .GroupBy(p => p.UserId)
+                .Where(g => g.Count() == 1 && g.All(p => p.IsOfficial))
+                .SelectMany(gr => gr.ToList());
```

Comment:
Since we know that we have only one group, we can use `.Select(gr => gr.FirstOrDefault());`. This will be faster than `SelectMany()`